### DISCUSSION TITLE
linux-experience-part1/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#22] [Announcements]
# linux-experience-part1/

پست قبلیم که در مورد لینوکس گذاشتم ثقیل و سنگین بود. این پست رو سعی می‌کنم روان‌تر بنویسم. بریم ببینیم که چی پیش میاد.
پست قبلی: توزیع‌های مختلف لینوکس - ارائه یک دسته‌بندی | گاه‌نوشته‌های امیرپورمند
اولین آشنایی من با لینوکس در درس آزمایشگاه سیستم‌عامل بود. اون زمان با Ubuntu 18.04 کار می‌کردیم. راستش از اون زمان خیلی از دستکاپ اوبونتو (Gnome) خوشم نیومد. اصلا درک نمی‌کردم چرا کسی باید ویندوز رو ول کنه و با لینوکس کار کنه.

https://aprd.ir/linux-experience-part1/

**Comment by amirghh** [comment]
سلام
من رشته و تخصص کامپیوتر نیست اما علاقمند هستم برای یکبار هم شده غیر از ویندوز سیستم عامل دیگه ایی رو تجربه کنم. ولی همیشه نبود اطلاعاتی مثل اینکه چه سخت افزاری جوابگو هست(مثلا الان روی لپ تاپم ویندوز10 دارم) و اینکه نرم افزارهایی که الان دارم استفاده میکنم مثل آفیس، اسکایپ، تلگرام و … آیا براحتی برای لینوکس هم وجود داره؟ نصب اون به چه شکل هست و سوالاتی از این قبیل مانع از نصب لینوکس برام می شد(ویندوز رو دیگه یاد گرفتیم موقع نصب چطور یمشکلاتش رو برطرف کنیم :-) ). خوشحال میشم بیشتر بخونم خصوصا اینکه تفاوت در حس کار کردن با این دو توریع متفاوت رو بیشتر باز کنی ممنون می شم.

پست قبلیت هم شاید کمی گیج کننده بود ولی برای کسی که دنبال آشنایی با این دنیا باشه به نظرم اگر آهسته خونده بشه خیلی هم خوبه

**Reply by pourmand1376** [reply]
سلام. وقت بخیر. 

راستش در ادامه پست می‌خواستم اصلاً به همین برسم که آیا لینوکس می‌تونه تمام نیازهای ما رو مرتفع کنه؟ جوابی که من بهش رسیدم اینه که بستگی داره نیازت چی باشه. لینوکس همه نرم‌افزارهایی که گفتی رو داره. 

اما تو چند تا موضوع ضعف داره. یکی این که لینوکس برای بازی کردن نیست. در واقع اگر بازی می‌کنی مک هم چندان به دردت نخواهد خورد. 
دوم این که یک سری نرم‌افزارهای خاص مثل سری آفیس (ورد و اکسل و پاورپوینت) و یا نرم‌افزارهای Adobe رو لینوکس نداره. 

البته می‌دونم کلی Alternative هست. ولی خودمونیم. من همه نرم‌افزارهایی که Alternative ورد هستند رو تست کردم. هیچ کدوم انگشت کوچیکه ورد هم نمیشن. با همه اوصاف با اینها میشه کنار اومد. من یکی دو سالی اصلاً ویندوز استفاده نکردم و خب بالاخره کارهام رو با هر بدبختی‌ای بوده انجام دادم. 

یک سری نرم‌افزارها هم هستند که اصلاً برای لینوکس نمیان. برای مک هم ممکنه نیان. مثلاً قندشکنی که استفاده میکنم نسخه ویندوزش خیلی عالیه. رو لینوکس هم میشه یه جورایی وصل شد ولی انقدر خوب نیست.

حالا جلوتر بریم بیشتر هم توضیح میدم. ولی فکر می‌کنم تجربه لینوکس در کل ارزشمند هست. برای برنامه‌نویسی عالی هست و تجربه خیلی خوبی ازش دارم. ولی برای کاربردهای عادی (مثل فیلم دیدن و ورد کار کردن و خیلی کارهای عادی دیگه) ویندوز و مک خیلی راحت ترن. ابزارهایی که براشون وجود داره حرفه‌ای تره و وقت آدم برای چیزهای الکی تلف نمیشه. 

مثلاً بگم اگر ویندوز داشته باشی یه PotPlayer نصب می‌کنی و هر فیلمی رو می‌بینی. تموم شد رفت. 
لینوکس هم میشه نرم‌افزار نصب کرد. ولی یه بار میگه ffmpeg نیست. یه بار میگه codec های غیراستاندارد هم نصب کن برام. بعد این که تو لینوکس اکثر پلیرها با دو زیرنویس همزمان (یکی بالا و یکی پایین) مشکل دارن. خلاصه بعضی وقت‌ها دردسر داره. 

یکی دوستان من می‌گفت ویندوز و مک خوبن چون انقدر Userbase شون زیاده که در اکثر موارد «It just works». همین میشه که اکثر نرم‌افزارها میگن First-class citizen ما ویندوز و مک هست. حالا اشکال نداره. لینوکس هم ساپورت می‌کنم اما در اولویت دوم!

**Comment by hiwasham** [comment]
مرسی امیر جان،
برای منی که اولین برخوردم با لینوکس سال 83 بود، خوندن این پست بهم یادآوری کرد که چقدر سنم بالا رفته. 
من اون موقع اوپن سوز رو لپتاپم داشتم
ولی چند سال بعد، به خاطر راحتی و دسترسی سریعتر به بعضی ابزارهای کرک شده، ویندوز رو هم کنار لینوکس نصب کردم. اون زمان روم نمیشد به دوستام بگم ویندوز هم دارم. خودت میدونی فضای آکادمیک کامپیوتری و لینوکس و جامعه آزاد و استالمن و اینها چطوریه
ما اون زمان کمی درگیر برگزاری ایونت های لاگ هم شدیم
LUG(Linux user group)
بعد از چند سال کلاً لینوکس رو کنار گذاشتم
الان بعد از هفت هشت سال دوری از لینوکس، مدتیه نمیدونم چرا ولی دلم برای لینوکس تنگ شده
کنجکاو هم هستم ببینم چقدر تغییر کرده
این نوشته هات رو تبدیل میکنم به یه فرصت و سعی میکنم کم کم به این فضا برگردم

**Reply by pourmand1376** [reply]
سلام هیوا جان، 

ماشاءالله. پس شما از قدیم باهاش کار می‌کردید. 

منم لینوکس رو دوست دارم. البته فکر نمی‌کنم تجربه دستکاپ بهتری از ویندوز و مک برای کاربر بده. 

ولی ایده‌هاش خیلی جذابن. همین که ما آدم ها تونستیم کنار هم چیزی درست کنیم که شرکت‌های غول تکنولوژی نتونستن هنوز رو دستش چیزی بیارن ارزشمنده. شما بیشتر از من می‌دونید که الان در زمینه سرور میشه گفت انتخاب اول اکثر قریب به اتفاق سرورها لینوکس هست و کلاً بازار رو گرفته. 

هر چند هنوز به نظرم برای دستکاپ خیلی جای کار داره. 

ولی همین که یک سری آدم نشستن آخر هفته ها وقت گذاشتن و محصولی درست کردند که از یک کمپانی Multi-Billionare - به نام مایکروسافت - جلو زده واقعا ارزشمنده. همین بیل گیتس یه زمانی گفته بود که اینهایی که نرم‌افزار مجانی درست می‌کنند احمقی بیش نیستند ولی طولی نکشید که همون احمق‌ها کارهایی رو کردند که مایکروسافت فهمید دیگه حریف‌شون نمیشه و خب گیت‌هاب رو خرید و کلی پروژه‌های خودش رو هم متن باز کرد. 

در کل منم خیلی به استالمن و جامعه آزاد و این‌ها علاقه دارم و گاه و بیگاه حرف‌هاشون رو دنبال می‌کنم.
